internal/resources/providers/gcplib/inventory/provider.go

```diff
@@ -150,60 +144,62 @@ func (p *ProviderInitializer) Init(ctx context.Context, log *logp.Logger, gcpCon
 	if err != nil {
 		return nil, err
 	}
+
+	displayNamesCache := NewMapCache[string]()
 	// wrap the resource manager client for mocking
 	crmServiceWrapper := &ResourceManagerWrapper{
 		getProjectDisplayName: func(ctx context.Context, parent string) string {
-			prj, err := crmService.Projects.Get(parent).Context(ctx).Do()
-			if err != nil {
-				log.Errorf("error fetching GCP Project: %s, error: %s", parent, err)
-				return ""
-			}
-			return prj.DisplayName
+			return displayNamesCache.Get(func() string {
+				prj, err := crmService.Projects.Get(parent).Context(ctx).Do()
+				if err != nil {
+					log.Errorf("error fetching GCP Project: %s, error: %s", parent, err)
+					return ""
+				}
+				return prj.DisplayName
+			}, parent)
 		},
 		getOrganizationDisplayName: func(ctx context.Context, parent string) string {
-			org, err := crmService.Organizations.Get(parent).Context(ctx).Do()
-			if err != nil {
-				log.Errorf("error fetching GCP Org: %s, error: %s", parent, err)
-				return ""
-			}
-			return org.DisplayName
+			return displayNamesCache.Get(func() string {
+				org, err := crmService.Organizations.Get(parent).Context(ctx).Do()
+				if err != nil {
+					log.Errorf("error fetching GCP Org: %s, error: %s", parent, err)
+					return ""
+				}
+				return org.DisplayName
+			}, parent)
 		},
 	}
 
 	return &Provider{
-		config:    gcpConfig,
-		log:       log,
-		inventory: assetsInventoryWrapper,
-		crm:       crmServiceWrapper,
-		crmCache:  make(map[string]*fetching.CloudAccountMetadata),
+		config:                    gcpConfig,
+		log:                       log,
+		inventory:                 assetsInventoryWrapper,
+		crm:                       crmServiceWrapper,
+		cloudAccountMetadataCache: NewMapCache[*fetching.CloudAccountMetadata](),
```

Comment:
Is `Provider` lifecycle per-cycle or long-lived? I think it is long-lived its initialized once at the startup of cloudbeat, but perhaps I am mistaken here. 

If it is, then shouldn't we be extra careful what we cache once and only? Is cloud account metadata safe to retrieve only once per lifetime? If not we should use one of the many in-mem cache libraries that implements ttl global or per key ttl.

Reply:
before this PR, the cache was a plain map:
```diff
- crmCache map[string]*fetching.CloudAccountMetadata
+ cloudAccountMetadataCache *MapCache[*fetching.CloudAccountMetadata]
```

so it's still the same behavior as before, only now a little less repetitive as `MapCache` just takes away the operations we'd need to do to cache values using a plain `map[string]` 

in general though, i agree this behaviour is probably not correct, even though project/org names probably rarely change, we'd probably still want to get fresh values. i've opened an [issue](https://github.com/elastic/cloudbeat/issues/2182) to address this in a separate PR (as the behaviour itself didn't change from before and is unrelated to this PR)